resolwe_bio/processes/microarray/methylation_arrays.py

```diff
@@ -0,0 +1,128 @@
+    category = "Methylation arrays"
+    data_name = 'SeSAMe array ({{ idat_file.red_channel.file|default("?") }})'
+    scheduling_class = SchedulingClass.BATCH
+    persistence = Persistence.CACHED
+    entity = {"type": "sample"}
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {
+                "image": "public.ecr.aws/s4q6j6e8/resolwebio/methylation_arrays:1.0.0"
+            }
+        },
+        "resources": {
+            "cores": 8,
+            "memory": 16384,
+        },
+    }
+
+    class Input:
+        """Input fields to process MethylationArraySesame."""
+
+        idat_file = DataField(
+            data_type="methylationarray:idat",
+            label="Illumina methylation array IDAT file",
+            description="Illumina methylation array BeadChip raw IDAT file.",
+        )
+
+    class Output:
+        """Output fields to process MethylationArraySesame."""
+
+        methylation_data = FileField(label="A gzipped tab delimited file (txt.gz)")
+        qc_data = FileField(label="Quality control information from SeSAMe analysis")
+        species = StringField(label="Species")
+        platform = StringField(label="Platform used in the analysis")
+
+    def run(self, inputs, outputs):
+        """Run MethylationArraySesame process."""
+
+        dirdata = Path("./data")
+        if not dirdata.exists():
+            dirdata.mkdir()
+
+        red = inputs.idat_file.output.red_channel.path
+        green = inputs.idat_file.output.green_channel.path
+        [copy2(src=x, dst=dirdata.name) for x in [red, green]]
+
+        platform = inputs.idat_file.output.platform
+        manifest = f"{platform}.hg38.manifest"
+
+        sesame_args = [
+            f"--platform={platform}",
+            f"--manifest={manifest}",
+        ]
+        rc, _, _ = Cmd["sesame.R"][sesame_args] & TEE(retcode=None)
+        # Returns QC_data.txt and beta_values_annotated.txt.gz
+
+        if rc:
+            self.error(
+                "An error was encountered during the running of SeSAMe pipeline."
+            )
+
+        meth_data = "beta_values_annotated.txt.gz"
+        xy = pd.read_csv(
+            filepath_or_buffer=meth_data,
+            sep="\t",
+        )
+
+        mapping_filters = {"source_db": f"ILLU_{platform}"}
+        mappings = self.mapping.filter(**mapping_filters)
+        ensembls = []
+        for mapping in mappings:
+            ensembls.append(
+                {"source_id": mapping.source_id, "target_id": mapping.target_id}
+            )
+
+        ensembls = pd.DataFrame(ensembls)
+
+        out = xy.merge(ensembls, left_on="probe_ids", right_on="source_id", how="left")
+        out = out[["probe_ids", "betas", "mvals", "pvals", "target_id"]]
+        out.columns = ["probe_ids", "ensembl_id", "betas", "mvals", "pvals"]
+
+        out.to_csv(
+            path_or_buf=meth_data,
+            sep="\t",
+            compression="gzip",
```

Comment:
Perhaps `index=False` could be used for the output file. IMO the row index is not relevant here.